apps/golden-sample-app/src/app/app.module.ts

```diff
@@ -58,23 +71,44 @@ import { CookieService } from "ngx-cookie-service";
     {
       provide: APP_INITIALIZER,
       multi: true,
-      deps: [ OAuthService, CookieService ],
-      useFactory: (oAuthService: OAuthService, cookieService: CookieService) => () => {
-        // Remove this if auth cookie is not needed for the app
-        oAuthService.events.subscribe(e => {
-          if (e.type === 'token_received' || e.type === 'token_refreshed') {
-            // Set the cookie on the app domain
-            cookieService.set('Authorization', `Bearer ${ oAuthService.getAccessToken() }`);
-          }
-        });
+      deps: [OAuthService, CookieService],
+      useFactory:
+        (oAuthService: OAuthService, cookieService: CookieService) => () => {
+          // Remove this if auth cookie is not needed for the app
+          oAuthService.events.subscribe(({ type }) => {
+            if (type === 'token_received' || type === 'token_refreshed') {
+              // Set the cookie on the app domain
+              cookieService.set(
```

Comment:
how would this work if we would like a refresh token?

Reply:
The event should fire again and the cookie overwritten when the access token is refreshed